android/build.gradle

```
@@ -2,50 +2,20 @@
 
 buildscript {
     ext {
-        buildToolsVersion = "31.0.0"
+        buildToolsVersion = "33.0.0"
         minSdkVersion = 21
-        compileSdkVersion = 31
-        targetSdkVersion = 31
+        compileSdkVersion = 33
+        targetSdkVersion = 33
 
-        if (System.properties['os.arch'] == "aarch64") {
-            // For M1 Users we need to use the NDK 24 which added support for aarch64
-            ndkVersion = "24.0.8215888"
-        } else {
-            // Otherwise we default to the side-by-side NDK version from AGP.
-            ndkVersion = "21.4.7075529"
-        }
+        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
```

The old comment said that M1 support was added in 24. I checked, and that's correct. 23 works non-natively on M1 via Rosetta translation.

I wasn't familiar with "side by side" and did a bit of reading. If I understood correctly, that's a term for when you use different versions of the NDK for different modules in the same project. AFAICT, that's not happening here. (We only have one module, right? Or does ReactNative itself include multiple modules with different versions?)

All that said, I think it would be better to say:
```suggestion
        // We use NDK 23 which is the latest LTS release.
```

If that's not the reason, then I suggest we try 24 -- which, with native M1 support, probably builds faster!

---

This code was updated by react-native during the upgrade to 0.71.1. You can see the change in the upgrade helper: 
https://react-native-community.github.io/upgrade-helper/?from=0.70.5&to=0.71.1

---

Should we make a PR in the react-native repo?

---

By the way, this reminds me that the android workflow of this pipeline probably should build on the Linux stack and not on Xcode for the performance and stability benefits.

Regarding the original issue: I don't recommend deviating from the NDK version of React Native, I'm sure there is a good reason they target one version and not the other, debugging C compilation issues in a Gradle build is not a fun thing 😬 Also, each Android Gradle Plugin version only supports a specific range of NDK versions, so it's best to stick to the defaults. And by running the Android workflow on Linux, the M1 issue becomes irrelevant.

---

> By the way, this reminds me that the android workflow of this pipeline probably should build on the Linux stack and not on Xcode for the performance and stability benefits.

28734b4